src/main/java/com/michelin/ns4kafka/services/executors/TopicAsyncExecutor.java

```diff
@@ -391,36 +473,62 @@ private void createAndAssociateTags(List<TagTopicInfo> tagsToAssociate) {
         schemaRegistryClient.createTags(tagsToCreate, managedClusterProperties.getName())
             .subscribe(
                 successCreation ->
-                    schemaRegistryClient.associateTags(
-                            managedClusterProperties.getName(),
-                            tagsToAssociate)
-                        .subscribe(
-                            successAssociation ->
-                                log.info(String.format("Success associating tag %s.", stringTags)),
-                            error ->
-                                log.error(String.format("Error associating tag %s.", stringTags), error)),
+                    schemaRegistryClient.associateTags(managedClusterProperties.getName(), tagsToAssociate)
+                    .subscribe(
+                        successAssociation -> {
+                            ns4kafkaTopics.stream()
+                                .filter(topic -> tagsToAssociate
+                                    .stream()
+                                    .map(TagTopicInfo::entityName)
```

Comment:
I agree, so I built the Mapping withh your suggestion and use this mapping for the loop for status update and logs.

Your assumption seems to be correct, topics without new tags don't need to be updated, so filter can drastically reduce the map size. The filter condition should just be inverted but otherwise, it works